docs/DevGuide/BuildSystem.md

```diff
@@ -51,9 +110,72 @@ or a block can be excluded using `LCOV_EXCL_START` and `LCOV_EXCL_STOP`.
 However, this should be used extremely sparingly since unreachable code paths
 should be removed from the code base altogether.
 
-## Adding Executables
+## Adding Executables {#adding_executables}
+
+All general executables are found in `src/Executables`, while those
+for specific evolution (elliptic) systems are found in
+`src/Evolution/Executables` (`src/Elliptic/Executables`).  See \ref
+dev_guide_creating_executables "how to create executables".
+
+## Adding External Dependencies {#adding_external_dependencies}
+
+To add an external dependency, first add a `SetupDEPENDENCY.cmake`
+file to the `cmake` directory. You should model this after one of the
+existing ones for `Catch` or `Brigand` if you're adding a header-only
+library and `yaml-cpp` if the library is not header-only. If CMake
+does not already support `find_package` for the library you're adding
+you can write your own. These should be modeled after `FindBrigand` or
+`FindCatch` for header-only libraries, and `FindYAMLCPP` for compiled
+libraries. The `SetupDEPENDENCY.cmake` file must then be included in
+the root `spectre/CMakeLists.txt`. Be sure to test both that setting
+`LIBRARY_ROOT` works correctly for your library, and also that if the
+library is required that CMake fails if the library is not found.
 
-All general executables are found in `src/Executables`. To add a new executable
-add a directory for it, and that subdirectory to
-`src/Executables/CMakeLists.txt` and then model the `CMakeLists.txt` and other
-files after one of the existing executables.
+## Commonly Used CMake flags {#common_cmake_flags}
+The following are the most common flags used to control building with
+`CMake`. They are used by
+```
+cmake -D FLAG1=OPT1 ... FLAGN=OPTN <SPECTRE_ROOT>
```

Comment:
Add another `-D` before `FLAGN` so it is clear that it has to be repeated for each.